### ISSUE TITLE
Loss becomes NAN after 30000 steps.

### ISSUE THREAD

**Issue by hengfei-wang** [Dong142857/Live3DPortrait#9]
Hi,

I am running the code on my datasets. I found that loss becomes NAN after around 30000 steps. Did you have the same problem? 

Thanks!

**Comment by Dong142857** [comment]
Yes,,,, this loss is easy to become NAN after adding GAN Loss. You can try to add gan loss after a long time warmup, set  'add_gan_loss_step' to about 20,000 iterations. In addition, there may be another two ways to alleviate this problem in my experiment:
1. Increase the r1_gamma coefficient.
2. Gradient clipping trick.

In fact, this NAN problem has not been completely solved, but is will hardly appear after these attempts.

**Comment by hengfei-wang** [comment]
Thank you for the advice.

I did not use GAN loss when training on my dataset. But the NAN problem always happens after a while. I check the input data. It seems ok. I tried gradient clipping trick with a max value of 0.5. But it does not work. Still confused :(

regards

**Comment by Dong142857** [comment]
Only reconstuction loss should not cause NAN problem. This experiment should have a warmup period. I suddently realized that i trained the renderer of eg3d before training the encoder using this code [#L277](https://github.com/Dong142857/Live3DPortrait/blob/85e43b403baf43284d28dbada48b634fe8ac712f/training/coach.py#L277) . Do you directly train the encoder? 

Since this code was completed  a long time ago, I have forgotten some training details. However,  the main steps should be as follows:

1. Sample triplanes to render the high quality images. Train only the renderer. 
2. Fix renderer and use reconstruction loss to warm up the encoder training process.
3. Add GAN Loss and joinly train the render after generating reasonable result. Note that the step of adding gan loss and lauching renderer training can be different.

Sorry again for my messy code. I have no time to refresh it.  If you encounter any further training issues, please let me know and I will try to resolve them as quickly as possible.
